src/main/java/se/sundsvall/partyassets/service/AssetService.java

```diff
@@ -0,0 +1,74 @@
+package se.sundsvall.partyassets.service;
+
+import static org.zalando.problem.Status.CONFLICT;
+import static org.zalando.problem.Status.NOT_FOUND;
+import static se.sundsvall.partyassets.integration.db.model.PartyType.ENTERPRISE;
+import static se.sundsvall.partyassets.integration.db.model.PartyType.PRIVATE;
+import static se.sundsvall.partyassets.integration.db.specification.AssetSpecification.createAssetSpecification;
+import static se.sundsvall.partyassets.service.mapper.AssetMapper.toEntity;
+import static se.sundsvall.partyassets.service.mapper.AssetMapper.updateEntity;
+
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+import org.zalando.problem.Problem;
+
+import se.sundsvall.partyassets.api.model.Asset;
+import se.sundsvall.partyassets.api.model.AssetCreateRequest;
+import se.sundsvall.partyassets.api.model.AssetSearchRequest;
+import se.sundsvall.partyassets.api.model.AssetUpdateRequest;
+import se.sundsvall.partyassets.integration.db.AssetRepository;
+import se.sundsvall.partyassets.integration.db.model.PartyType;
+import se.sundsvall.partyassets.integration.party.PartyClient;
+import se.sundsvall.partyassets.service.mapper.AssetMapper;
+
+@Service
+public class AssetService {
+
+	private final AssetRepository repository;
+
+	private final PartyClient partyClient;
+
+	public AssetService(final AssetRepository repository, final PartyClient partyClient) {
+		this.repository = repository;
+		this.partyClient = partyClient;
+	}
+
+	public List<Asset> getAssets(AssetSearchRequest request) {
+		return repository.findAll(createAssetSpecification(request))
+			.stream()
+			.map(AssetMapper::toAsset)
+			.toList();
+	}
+
+	public String createAsset(final AssetCreateRequest request) {
+		if (repository.existsByAssetId(request.getAssetId())) {
+			throw Problem.builder()
+				.withStatus(CONFLICT)
+				.withTitle("Asset already exists")
+				.withDetail("Asset with assetId %s already exists".formatted(request.getAssetId()))
+				.build();
+		}
+		return repository.save(toEntity(request, calculatePartyType(request.getPartyId()))).getId();
+	}
+
+	private PartyType calculatePartyType(String partyId) {
+		return partyClient.getLegalId(generated.se.sundsvall.party.PartyType.PRIVATE, partyId)
+			.isPresent() ? PRIVATE : ENTERPRISE;
```

Comment:
If "dummy" is sent to this method, "ENTERPRISE" will be returned. Is this an appropriate behaviour?

Reply:
This attribute is not requested by the stakeholders and is not visible in the API, it's more of a "good to have" thing. Throwing exception for non existing partyIds (which will probably never happen) might not be aligned with the requirements from the stakeholders.

Another option is to add a third partyType, UNKNOWN, if id isn't recognized as private or enterprise type.

I'll check with Jakob on how to handle if/when type for partyId is not recognized. TBC.

Reply:
Solution is to go with your recommendation and throw fault if party type can not be calculated, this is now changed.